### PR TITLE
Comarch product connector

### PR DESCRIPTION
**Pull request by rowinskidamian**
Working classes to get productList and product by id from comarch api.
Please add review.

### CODE REVIEW

File: fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/connectors/ComarchApiProductConnector.java [hunk 1]
```diff
@@ -1,4 +1,44 @@
...
+@Component
 public class ComarchApiProductConnector {
+
+    private HttpConnectorUtils httpConnectorUtils;
+    private ComarchProductConverter comarchProductConverter;
+    private ComarchProductMapper comarchProductMapper;
+
+    private final String getAllEndpoint = "https://app.erpxt.pl/api2/public/products";
+    private final String getByIdEndpoint = "https://app.erpxt.pl/api2/public/products/";
+
```
Comment (KrzKos) [inline fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/connectors/ComarchApiProductConnector.java:22]: In HttpConnectorUtils method httpGetById has modifier for url (url + "/" + id). It is not necessary to set two endpoints (without and with "/").

File: fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/connectors/ComarchApiProductConnector.java [hunk 2]
```diff
@@ -1,4 +1,44 @@
...
+    public List<ComarchProductDTO> getAllProductList () throws IOException {
+        Response response = httpConnectorUtils.httpGetAll(getAllEndpoint);
+        String responseString = response.body().string();
+        List<ComarchProduct> comarchProductList = comarchProductConverter.fromList(responseString);
+        return comarchProductMapper.fromList(comarchProductList);
+    }
+
+    public ComarchProductDTO getProductById(Long id) throws IOException {
+        Response response = httpConnectorUtils.httpGetById(getByIdEndpoint, id);
```
Comment (KrzKos) [inline fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/connectors/ComarchApiProductConnector.java:38]: Dose it work? httpGetById(url, id) should accepts url without "/" at the end. It adds "/" and id inside the method. In your case result will be i.e https://app.erpxt.pl/api2/public/products//21340 with double "/" before 21340.

File: fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/dto/ComarchProductDTO.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package pl.fakturogen.comarch.connector.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (KrzKos) [inline fakturogen-comarch-connector/src/main/java/pl/fakturogen/comarch/connector/dto/ComarchProductDTO.java:9]: Can we use Lombok annotation for constructor?

**Review by KrzKos — CHANGES_REQUESTED** [review]
Please see my comments.

File: fakturogen-web/src/main/java/pl/fakturogen/web/controller/rest/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+@RestController
+public class ProductController {
+    private ComarchApiProductConnector comarchApiProductConnector;
+
+    public ProductController(ComarchApiProductConnector comarchApiProductConnector) {
+        this.comarchApiProductConnector = comarchApiProductConnector;
+    }
+
+    @RequestMapping("/getProductList")
```
Comment (ewa-git) [inline fakturogen-web/src/main/java/pl/fakturogen/web/controller/rest/ProductController.java:19]: Maybe in REST controllers would be nice to use @GetMapping, @PostMapping etc.

**Review by ewa-git — APPROVED** [review]
only small remarks

**Pull request merged**